### PR TITLE
tctl support for DynamicWindowsDesktop

### PR DESCRIPTION
**Pull request by probakowski**
This is part of splitting https://github.com/gravitational/teleport/pull/46738 into smaller, easier to review, PRs

changelog: Add support for DynamicWindowsDesktop to tctl

### CODE REVIEW

**Comment by probakowski** [comment]
@zmb3 @fspmarshall @rudream friendly ping

**Review by fspmarshall — APPROVED** [review]

**Review by rosstimothy — COMMENTED** [review]
Can we update [TestEditResources](https://github.com/gravitational/teleport/blob/master/tool/tctl/common/edit_command_test.go#L49) and [TestCreateResources](https://github.com/gravitational/teleport/blob/master/tool/tctl/common/resource_command_test.go#L1360) with cases for this resource?

File: tool/tctl/common/resource_command.go [hunk 1]
```diff
@@ -891,6 +893,45 @@ func (rc *ResourceCommand) createWindowsDesktop(ctx context.Context, client *aut
...
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	dynamicDesktopClient := client.DynamicDesktopClient()
+	if _, err := dynamicDesktopClient.CreateDynamicWindowsDesktop(ctx, wd); err != nil {
+		if trace.IsAlreadyExists(err) {
+			if !rc.force {
+				return trace.AlreadyExists("application %q already exists", wd.GetName())
+			}
+			if _, err := dynamicDesktopClient.UpdateDynamicWindowsDesktop(ctx, wd); err != nil {
```
[the remark this answers is not in the document]
Comment (probakowski) [reply tool/tctl/common/resource_command.go:907]: The same pattern is used for other resources like apps and kube clusters. We can't do much better unless we want to expose upsert function

[the remark this answers is not in the document]
Comment (zmb3) [reply tool/tctl/common/resource_command.go:907]: What's the alternative, Tim?

**Review by zmb3 — APPROVED** [review]

File: tool/tctl/common/resource_command.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (probakowski) [reply tool/tctl/common/resource_command.go:907]: I'll merge this as is and add upsert and use it here in follow-up PR

**Pull request merged**